velox/dwio/common/ColumnSelector.h

```diff
@@ -301,6 +316,27 @@ class ColumnSelector {
   // get node ID list to be read
   std::vector<uint64_t> getNodeFilter() const;
 
+  void checkSelectColNonDuplicate(bool readAsLowerCase) {
+    if (!readAsLowerCase) {
+      return;
+    }
+    std::unordered_map<std::string, int> names;
+    for (auto node : nodes_) {
+      auto& name = node->getNode().name;
+      if (names.find(name) == names.end()) {
+        names[name] = 1;
```

Comment:
`names[name]` will default set to 0, so you don't need the `if` to detect if it already exists or not